.github/workflows/test-pr-macos.yml

```diff
@@ -0,0 +1,156 @@
+name: "Test PR on MacOS"
+on:
+  pull_request:
+    types: [opened, edited, reopened, synchronize]
+    branches:
+      - "develop"
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  version-sync:
+    name: "Version Sync"
+    runs-on: ubuntu-20.04
+    steps:
+      - name: "Check out code"
+        uses: actions/checkout@v3
+        with:
+          token: ${{ secrets.JENKINS_GITHUB_PAT }}
+          # fetch-depth 0 means deep clone the repo
+          fetch-depth: 0
+      - name: "Update Version"
+        run: |
+          set -x
+          git config user.name devops
+          git config user.email devops@runtimeverification.com
+          ./package/version.sh sub
+          if git add --update && git commit --message "Set Version: $(cat package/version)"; then
+            git push origin HEAD:${GITHUB_HEAD_REF}
+          fi
+
+  nix-maven:
+    name: "Maven Build"
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - runner: macos-13
+            os: macos-13
+          - runner: MacM1
+            os: self-macos-12
+    runs-on: ${{ matrix.runner }}
+    needs: version-sync
+    steps:
+      - name: "Check out code, set up Git"
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+          fetch-depth: 0
+      - name: "Upgrade bash"
+        if: ${{ contains(matrix.os, 'macos') }}
+        run: brew install bash
+      - name: "Install Dependencies"
+        run: brew install bison boost cmake flex fmt gcc gmp openjdk jemalloc libyaml llvm@15 make maven mpfr pkg-config python stack zlib z3
+      - name: "Install and set up direnv"
+        run: |
+           set -euxo pipefail
+           export PATH="$(brew --prefix bison)/bin:$PATH"
+           export PATH="$(brew --prefix flex)/bin:$PATH"
+           export PATH="$(brew --prefix llvm@15)/bin:$PATH"
+           export PATH="$(brew --prefix openjdk)/bin:$PATH"
+           echo $(which llvm-config)
+           export LLVM_CONFIG=$(which llvm-config)
+           export LLVM_DIR="$(brew --prefix llvm@15)/lib/cmake/"
+           mvn package -T1C
+         # brew install direnv
+         # cp macos-envrc .envrc
+         # eval "$(direnv hook bash)"
+         # direnv allow .
```

Comment:
Looks like you're reaching that conclusion, but `direnv` is not something I'd expect to see in CI - it's just a local developer convenience tool.

Reply:
Actually, I don't quint follow what you mean by
> the Homebrew formula is in general a pretty good source for the best way to build K from source on macOS

Could you please elaborate on that?

Reply:
The code is here: https://github.com/kframework/homebrew-k/blob/master/Formula/kframework.rb

If you've not seen a Homebrew formula before, it's basically a Ruby DSL that allows you to specify dependencies and build steps for a package. We maintain one for K; if there are macOS-specific issues with the build then it's plausible that the Homebrew code has already dealt with it somehow